src/eventProcessor/GoogleAnalyticsEventProcessor.js

```diff
@@ -115,6 +115,27 @@ class GoogleAnalyticsEventProcessor {
     return 'googleAnalytics';
   }
 
+  /**
+   * Builds the POST request URL using `measurement_id` and `api_secret`
+   * query parameters if available
+   * @return {string}
+   * @private
+   */
+  buildRequestUrl() {
```

Comment:
I've found that Array.join can simplify URL query string building a lot when some or all of the parameters are optional.

e.g. psuedo-code like:
```
const paramsArray = [];
if (measurementId) paramsArray.push(`measurement_id=${measurementId}`);
// etc.

if (paramsArray.length > 0) {
  const params = paramsArray.join('&');
  // etc..
}
```